feat: POST CRUD 구현  [ISSUE-25]

jamie9504 opened the pull request:
**이슈 번호**

resolved: #25 

**작업 내용**

1. POST CRUD 작성
    - 인수테스트 작성
2. State Enum 추가
    - PUBLISHED : 정상 발행
    - DELETED : 삭제
    - 추후, 임시 저장 or 발행 대기 등으로 확장할 수 있음
    - 확장시 사용여부 YN, 발행여부 YN 등을 필드로 가질 수 있음
3. GlobalExceptionHandler 작성
    - IllegalArgumentException & Exception 처리
4. BaseEntity 분리
    - id, createdAt, modifiedAt 관리
    - 엔티티가 해당 추상클래스를 상속한다면, id, createAt, modifiedAt 필드를 별도로 추가하지 않아도 됨    

**테스트 작성 여부**

- [X] Test case

**참고 사항**
- 추후 논의할 부분
    - 일단 포스트 글자 제한 20자로 했는데, 이 부분은 회의 때 안건 재 상정
    - 해당 PR이 머지된 후 좋아요, 유저 연결, 사진 추가 등을 개발 진행
    - 추후 content (사진, 글 등) 개발할 때, 완전 공란 등의 validation 추가 작업 필요

jamie9504 committed:
feat: 글 등록 인수테스트 작성

- Mock Data를 반환하는 글 등록 컨트롤러 작성
- 글 작성 dto, 글 반환 dto 작성
- rest assured 의존성 추가

jamie9504 committed:
feat: 글 목록 조회 인수테스트 작성

- Mock Data를 반환하는 글 목록 조회 컨트롤러 작성

jamie9504 committed:
feat: 글 수정 인수테스트 작성

- Mock Data를 반환하는 글 수정 컨트롤러 작성
- PostCreateRequest를 PostRequest로 이름 변경

jamie9504 committed:
feat: 글 조회 인수테스트 작성

jamie9504 committed:
feat: 글 삭제 인수테스트 작성

- Mock Data를 반환하는 글 삭제 컨트롤러 작성

jamie9504 committed:
feat: 글 등록 서비스 작성

- 글 등록 서비스 Mock 테스트 작성
- PostRepository 작성
- Post Validation 메서드 및 테스트 작성
- PostController에 PostService 의존성 주입
- PostService에 PostRepository 의존성 주입

jamie9504 committed:
feat: 글 조회 서비스 작성

- 글 조회 서비스 (성공/예외) Mock 테스트 작성
- Displayname 적용

jamie9504 committed:
feat: Post 전체 목록 조회 서비스 및 테스트 작성

- PostResponse.listOf() 메서드 작성

jamie9504 committed:
feat: Post 전체 목록 조회 서비스 및 테스트 작성

jamie9504 committed:
feat: Post 삭제 서비스 및 테스트 작성

- Post에 Entity annotation 추가

jamie9504 committed:
refactor: BaseEntity 분리

- @MappedSuperclass를 사용하는 BaseEntity 분리 (Id, CreatedDate, LastModifiedDate)
- Test만을 위한 Id 세팅 생성자 제거

jamie9504 committed:
feat: State Enum 추가 및 ServiceTest 실제 DB로 변경

- PostServiceTest - Mock에서 실제 DB로 변경
- State Enum 추가
- Post에 State 필드 및 관련 메서드 추가
- 조회시 삭제상태 제외 기능 구현

jamie9504 committed:
refactor: 삭제한 ID로 재 삭제시 예외 발생 처리

jamie9504 committed:
feat: PostController Mock Data에서 PostService 호출로 변경

- PostController Mock Data에서 PostService 호출로 변경
- 인수테스트 주석 해제 및 실제 환경 테스트 완료
- GlobalExceptionHandler 추가
- ErrorResponse 객체 추가

jamie9504 committed:
refactor: @SpringBootTest > @DataJpaTest + @Import로 변경

jamie9504 committed:
chore: 롬복 PR merge

jamie9504 committed:
refactor: Enum 저장 방식 변경 (Ordinal > String)

ttkmw commented on back-end/legeno-around-here/src/main/java/wooteco/team/ittabi/legenoaroundhere/controller/GlobalExceptionHandler.java:
```diff
@@ -0,0 +1,24 @@
+public class GlobalExceptionHandler {
+
+    @ExceptionHandler(IllegalArgumentException.class)
+    public ResponseEntity<ErrorResponse> handleNotFound(IllegalArgumentException e) {
```
함수명은 handleNotFound인데, 잡는 익셉션은 IllegalArgumentException입니다.
로직상 NotFound 에러가 아닌데도 throw IllegalArgumentException을 하면 여기서 잡을 것 같은데, 맞다면 수정이 필요할 것 같습니다.

jamie9504 replied:
@ExceptionHandler에서 IllegalArgumentException을 잡아, NotFound로 처리해주고있습니다.

이 부분을 어떻게 수정할지는 다같이 얘기해보면 좋을 것 같은데요,
@ExceptionHandler Value에 익셉션을 설정해두면, 해당 익셉션은 이 메서드를 타게 되는데,

범용적인 익셉션이 아닌(결국 범용적인 익셉션의 경우, 분기 처리를 하지 않으면 한 Handler를 타게되니)
커스텀 익셉션을 사용하여 적절하게 예외를 분리하고, 분리한 예외별로 (혹은 그룹해서) HandlerMethod를 작성해야할 것 같습니다.

aegis1920 replied:
저도 제이미의 의견과 같습니다 :) IllegalArgumentException이 너무 범용적이라 어떻게 처리할 지 이야기하면 좋겠네요!

jamie9504 replied:
없는 리소스에 요청할 때(+삭제된 리소스)는 NotExistsException
사용자가 잘못된 값을 전달했을 때는 UserInputException를 주도록 바꿔봤습니다!

ttkmw commented on back-end/legeno-around-here/src/main/java/wooteco/team/ittabi/legenoaroundhere/controller/PostController.java:
```diff
@@ -0,0 +1,58 @@
+    }
+
+    @PostMapping
+    public ResponseEntity<Void> createPost(@RequestBody PostRequest postRequest) {
```
PostRequest가 조금은 다양하게 해석될 여지가 있을 것 같습니다.
PostRequest라고 하면 crud를 모두 포함하는 것 같거든요.
포괄적으로 dto를 만들 의도셨다면 PostDto는 어떨까요?
혹은 구체적인 의미를 담고 싶으시다면 CreatePostRequest는 어떨까요?

jamie9504 replied:
PostRequest / PostResponse로 DTO를 나누었는데요,
그 이유는 Request와 Response가 각각 담고있는 필드가 달랐기 때문입니다.

PostDto의 경우는, Request와 Response를 모두 아울러야할 것 같은 느낌이에요!

현재는 Create/Update가 모두 같은 필드만을 사용하기 때문에 PostRequest로 만들었는데,
추후 필요 필드가 달라지면 PostCreateRequest 등으로 분리하면 어떨까 싶습니다 :)

ttkmw commented on back-end/legeno-around-here/src/main/java/wooteco/team/ittabi/legenoaroundhere/domain/BaseEntity.java:
```diff
@@ -0,0 +1,59 @@
+    protected BaseEntity() {
+    }
+
+    public Long getId() {
```
Lombok 의존성 추가가 머지돼었으니, 롬복을 활용해보는 건 어떨까요?

jamie9504 replied:
Lombok은 저번 회의 때, 의존성은 추가하지만(SpringSecurity)
적용은 추후에 한 번에 리팩터링을 하자고 얘기가 나왔었어서 적용하지 않았습니다~! :)

ttkmw commented on back-end/legeno-around-here/src/main/java/wooteco/team/ittabi/legenoaroundhere/domain/BaseEntity.java:
```diff
@@ -0,0 +1,59 @@
+    }
+
+    @Override
+    public String toString() {
```
toString은 혹시 언제 쓰는건가요? 쓰는 장소가 없으면 없어도 될 것 같습니다.

jamie9504 replied:
toString의 경우 디버깅을 위해 보통 대부분의 객체에 재정의를 해주고 있습니다.

예전에 페어 프로그래밍할 때, 이펙티브 자바 12장을 참고해서 toString을 넣고나니 확실히 디버깅 때 한 눈에 볼 수 있어서 편하더라구요!

aegis1920 replied:
https://jaehun2841.github.io/2019/01/13/effective-java-item12/#%EC%84%9C%EB%A1%A0
여기를 읽어보셔도 좋을 것 같아요!

jamie9504 replied:
팀원 전체 toString 재정의로 컨벤션 가져가기로 했습니다.

ttkmw commented on back-end/legeno-around-here/src/main/java/wooteco/team/ittabi/legenoaroundhere/domain/Post.java:
```diff
@@ -1,5 +1,63 @@
+        this.state = State.PUBLISHED;
+    }
+
+    private void validateLength(String writing) {
```
private 함수는 public 함수보다 밑에 있어야 할 것 같습니다.

jamie9504 replied:
public 함수에서 추출한 private 함수의 경우 바로 그 아래에 둔다고 생각했는데요!

혹시 메서드 정렬 규칙이 따로 있는건지 궁금하네요!

aegis1920 replied:
https://stackoverflow.com/questions/4668218/are-there-any-java-method-ordering-conventions
메서드 순서의 목표는 코드를 쉽게 읽기 위함이니 회의할 때 이야기해봐도 좋을 것 같네요 :)

jamie9504 replied:
팀원 전체 회의를 통해, 호출하는 메서드 바로 아래에 놓기로 결정되었습니다.

ttkmw commented on back-end/legeno-around-here/src/main/java/wooteco/team/ittabi/legenoaroundhere/domain/Post.java:
```diff
@@ -1,5 +1,63 @@
+
+    private static final int LIMIT_LENGTH = 20;
+
+    private String writing;
```
writing 보다는 명사형이 좋지 않을까 합니아! content 는 어떨까요?

jamie9504 replied:
빙봉과 함께 찾아봤는데 writing 자체가 명사형이더라구요!
(저도 처음에 헷갈렸어요!)

content는 추후 writing 이외에 picture, voice ... 등등이 추가되면서 같이 추출하는 건 어떨까 싶었습니다 :)

ttkmw commented on back-end/legeno-around-here/src/main/java/wooteco/team/ittabi/legenoaroundhere/domain/Post.java:
```diff
@@ -1,5 +1,63 @@
+        }
+    }
+
+    public boolean isSameState(State state) {
```
현재 isSameState가 Deleted 상태인지 확인하는 부분에서만 쓰이는데, isDeleted를 만드는건 어떨까요?

jamie9504 replied:
이 부분은 처음에 isDeleted 식으로 생성하려고 하다가,
enum 확장에 따라서 점점 메서드가 늘어날 것 같다고 생각되어 isSameState로 변경하였던 부분인데,

그래도 isDeleted가 더 나을지 고민이네요!

aegis1920 replied:
추후에 Delete 뿐만 아니라 신고 여부, 임시 저장 등 여러 상태가 추가될 수 있기 때문에 isSameState 로 작성했습니다 :)

ttkmw commented on back-end/legeno-around-here/src/main/java/wooteco/team/ittabi/legenoaroundhere/domain/Post.java:
```diff
@@ -1,5 +1,63 @@
+        return Objects.equals(this.state, state);
+    }
+
+    public boolean isNotSameState(State state) {
```
이것도 isNotDeleted 어떤지요!

jamie9504 replied:
위에 댓글 달아두었습니다~!

ttkmw commented on back-end/legeno-around-here/src/main/java/wooteco/team/ittabi/legenoaroundhere/domain/Post.java:
```diff
@@ -1,5 +1,63 @@
+        return writing;
+    }
+
+    public void setWriting(String writing) {
```
혹시 jpa에 저장하려면 세터가 필수인가요??

jamie9504 replied:
이 부분 그렇게 알고 있었는데, 잘못 알고있었던 것 같아 확실히 알아본 후 수정하겠습니다~!

jamie9504 replied:
항상 필요한 부분이 아니라 필요 없는 부분 (Id 등)에서는 제거했습니다~!

ttkmw commented on back-end/legeno-around-here/src/main/java/wooteco/team/ittabi/legenoaroundhere/domain/State.java:
```diff
@@ -1,5 +1,6 @@
 
-public class State {
-
+public enum State {
```
domain 내에 post 패키지를 만든 후 그 안에서 State를 정의하거나, PostState라고 명시해주는 게 좋을 것 같습니다

jamie9504 replied:
post 뿐만아니라 comment도 State를 쓸 수 있을 것이라 생각하고 설계한 부분이었는데요!
조금 더 논의해보겠습니다~!

aegis1920 replied:
삭제되는 부분은 다른 곳(Comment, Sector 등)에서 쓰일 수 있기 때문에 범용적으로 네이밍했습니다!

ttkmw commented on back-end/legeno-around-here/src/main/java/wooteco/team/ittabi/legenoaroundhere/dto/ErrorResponse.java:
```diff
@@ -0,0 +1,21 @@
+
+public class ErrorResponse {
+
+    private String errorMessage;
```
ErrorResponse에 HttpStatus도 두는건 어떤가요?
클라이언트 입장에서 HttpStates도 알면 좋을 것 같아서요

jamie9504 replied:
HttpStatus의 경우엔, body로 보내지 않아도 사용할 수 있어서 별도로 보내지 않았습니다~!

ttkmw commented on back-end/legeno-around-here/src/main/java/wooteco/team/ittabi/legenoaroundhere/controller/PostController.java:
```diff
@@ -0,0 +1,58 @@
+
+    @PostMapping
+    public ResponseEntity<Void> createPost(@RequestBody PostRequest postRequest) {
+        Long postId = postService.createPost(postRequest).getId();
```
한 라인에서는 하나의 점만 사용하는게 좋지 않을까 합니다

jamie9504 replied:
짧은 메서드라 뒤에 붙였는데, 빙봉이랑 얘기 후 수정해보겠습니다~!

aegis1920 replied:
한 라인에 하나의 점만 사용하는 게 맞지만 get하는 부분은 디버깅하기 쉽고 가독성을 해치지 않는 것 같아서 유연하게 가져가도 된다고 생각했습니다 :)

jamie9504 replied:
- 이 부분은 가독성을 봤을 때는, 그대로 가는 것이 더 깔끔할 것 같아요!
``` diff
-         Long postId = postService.createPost(postRequest).getId();
+         Long postId = postService.createPost(postRequest)
+            .getId();
```

- 편의상 아래처럼, 사용하는 것처럼요 😉 
``` java
regions.keySet().stream()
```

ttkmw commented on back-end/legeno-around-here/src/main/java/wooteco/team/ittabi/legenoaroundhere/service/PostService.java:
```diff
@@ -0,0 +1,59 @@
+        return PostResponse.of(post);
+    }
+
+    private Post findNotDeletedPost(Long id) {
```
NotDeleted보다는 Published는 어떨까요?

ttkmw replied:
findPost가 있는데 findNotDeletedPost를 따로 둔 이유가 궁금합니다!

ttkmw replied:
다른 함수들은 리턴값이 PostResponse인데, 이 함수에서는 Post네요! 일관성있게 짜주는 게 좋지 않을까요?
개인적으로는 Post가 좋아보입니다

jamie9504 replied:
다른 함수들의 경우(public)엔 Controller와 이어지는 부분이라 PostResponse로 도출이 되었고
해당 함수의 경우(private)엔 그 함수들에서 중복 제거를 하기 위해 나온 부분이라 Post로 도출이 되었습니다.

일단 삭제된 글을 제외한 글을 조회한다는 개념으로 접근했기 때문에 NotDelete로 명명하였습니다.

aegis1920 replied:
함수를 명명하는 부분에 있어서 조금 더 고민해보겠습니다 :)

ttkmw commented on back-end/legeno-around-here/src/test/java/wooteco/team/ittabi/legenoaroundhere/acceptance/PostAcceptanceTest.java:
```diff
@@ -0,0 +1,153 @@
+    void managePost() {
+        // 등록
+        String expectedWriting = "글을 등록합니다.";
+        String location = createPost(expectedWriting);
```
url 혹은 location으로 이름을 통일하는 건 어떨까요?

jamie9504 replied:
지금 대부분 location으로 두고 있다고 생각했는데, 팀원 전체 통일을 말씀하시는걸까요?

ttkmw commented on back-end/legeno-around-here/src/test/java/wooteco/team/ittabi/legenoaroundhere/acceptance/PostAcceptanceTest.java:
```diff
@@ -0,0 +1,153 @@
+        // 삭제
+        deletePost(id);
+        findNotExistsPost(id);
+        List<PostResponse> postResponses2 = findAllPost();
```
postResponse2가 어떤 의미인지 잘 이해되지 않습니다
새로 선언하지 않고 기존에 쓰던 postResponse를 쓰거나, 의미를 담아주시면 좋겠습니다

jamie9504 replied:
앗, 수정한다고 생각했던 부분을 놓쳤네요! 알겠습니다~!

ttkmw commented on back-end/legeno-around-here/src/test/java/wooteco/team/ittabi/legenoaroundhere/service/PostServiceTest.java:
```diff
@@ -0,0 +1,124 @@
+        assertThat(postResponse.getWriting()).isEqualTo(createdPostResponse.getWriting());
+    }
+
+    @DisplayName("ID로 포스트 조회 - 실패")
```
부가설명을 더 적어주면 좋을 것 같습니다
존재하지 않는 ID로 포스트 조회 등으로요!

jamie9504 replied:
처음에 그렇게 적었다가, 삭제한 케이스를 제외하고 조회로 바뀌면서 빼버렸는데,(너무 길어지고, 빈번하게 바뀜)
 이 부분 다같이 있을 때 얘기해보면 좋을 것 같아요!

ttkmw commented on back-end/legeno-around-here/src/main/java/wooteco/team/ittabi/legenoaroundhere/domain/Post.java:
```diff
@@ -1,5 +1,63 @@
+@Entity
+public class Post extends BaseEntity {
+
+    private static final int LIMIT_LENGTH = 20;
```
20자면 넘 짧지 않을까여...?

jamie9504 replied:
PR에도 적어두었는데, 이 부분 다같이 모인 후 글자 수 협의하면 좋을 것 같아요!

ttkmw reviewed (commented):
수고하셨습니다!!

Hamliet commented on back-end/legeno-around-here/build.gradle:
```diff
@@ -19,6 +19,7 @@ dependencies {
     testImplementation('org.springframework.boot:spring-boot-starter-test') {
         exclude group: 'org.junit.vintage', module: 'junit-vintage-engine'
     }
+    testImplementation 'io.rest-assured:rest-assured:3.3.0'
```
버젼명을 없애는 것이 어떨까요?

jamie9504 replied:
버전명을 없앨 수 있군요!
4버전의 경우 약간 차이가 나긴 하던데 괜찮은가요?

aegis1920 replied:
없앨 경우 3.3.0이 되는 것을 확인했습니다 :) 👍

Hamliet commented on back-end/legeno-around-here/src/main/java/wooteco/team/ittabi/legenoaroundhere/controller/GlobalExceptionHandler.java:
```diff
@@ -0,0 +1,24 @@
+public class GlobalExceptionHandler {
+
+    @ExceptionHandler(IllegalArgumentException.class)
+    public ResponseEntity<ErrorResponse> handleNotFound(IllegalArgumentException e) {
```
User 관련 도메인에서 validation 과정에서 IllegalArgumentException을 던집니다. 루트가 이야기한 것처럼 함수명을 변경하는 것이 좋을 것 같습니다.

jamie9504 replied:
루트쪽에 댓글 달아둘께요~! 참고해주세요~!

Hamliet commented on back-end/legeno-around-here/src/main/java/wooteco/team/ittabi/legenoaroundhere/controller/PostController.java:
```diff
@@ -0,0 +1,58 @@
+    }
+
+    @PostMapping
+    public ResponseEntity<Void> createPost(@RequestBody PostRequest postRequest) {
```
회원 관리 쪽에서는 CRUD를 고려하여 CreateUserRequest와 같이 명명하였습니다. 참고하시길 바랍니다!

jamie9504 replied:
저희는 아직 C와 U가 동일하여 UserRequst로 명명했습니다.
CreateUserRequest와 UserCreateRequest 중에는 어떤 게 더 좋을지 논의해보면 좋을 것 같네요!

Hamliet commented on back-end/legeno-around-here/src/main/java/wooteco/team/ittabi/legenoaroundhere/controller/PostController.java:
```diff
@@ -0,0 +1,58 @@
+    @PostMapping
+    public ResponseEntity<Void> createPost(@RequestBody PostRequest postRequest) {
+        Long postId = postService.createPost(postRequest).getId();
+        return ResponseEntity.created(URI.create("/posts/" + postId)).build();
```
회원관리 쪽에서는
return ResponseEntity
.created(URI.create("/posts/" + postId))
.build();
로 나눴는데요, 이야기해서 하나의 컨벤션으로 맞추는게 좋을 것 같습니다.

jamie9504 replied:
이야기해보아요!

aegis1920 replied:
엔터를 치는 것이 가독성이 더 좋아보여서 저희도 적용했습니다! 👍

Hamliet commented on back-end/legeno-around-here/src/main/java/wooteco/team/ittabi/legenoaroundhere/domain/BaseEntity.java:
```diff
@@ -0,0 +1,59 @@
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
```
17라인, 20라인 공백도 User쪽에서는 붙였는데 이 부분도 논의하여 컨벤션을 통일하면 좋을 것 같습니다.

jamie9504 replied:
이 부분도 이야기해보아요!

jamie9504 replied:
엔티티는 엔터 치는 것으로 협의했습니다~!

Hamliet commented on back-end/legeno-around-here/src/main/java/wooteco/team/ittabi/legenoaroundhere/domain/BaseEntity.java:
```diff
@@ -0,0 +1,59 @@
+        return id;
+    }
+
+    public void setId(Long id) {
```
필요한 setter인가요?

jamie9504 replied:
😯 빠르게 알아본 후, 필요 없으면 지우겠습니다!

Hamliet commented on back-end/legeno-around-here/src/main/java/wooteco/team/ittabi/legenoaroundhere/domain/Post.java:
```diff
@@ -1,5 +1,63 @@
+@Entity
+public class Post extends BaseEntity {
+
+    private static final int LIMIT_LENGTH = 20;
```
저희는 MAX_LENGTH로 정의했는데요, 이 부분의 컨벤션도 이야기해보아요 :)

jamie9504 replied:
같이 얘기해봐요!

aegis1920 replied:
최소, 최대가 어울릴 것 같아 저희도 `MAX_LENGTH`, `MIN_LENGTH`로 반영했습니다! 👍

Hamliet commented on back-end/legeno-around-here/src/test/java/wooteco/team/ittabi/legenoaroundhere/acceptance/PostAcceptanceTest.java:
```diff
@@ -0,0 +1,153 @@
+            .body(params)
+            .contentType(MediaType.APPLICATION_JSON_VALUE)
+            .accept(MediaType.APPLICATION_JSON_VALUE)
+            .when()
```
논의해봐야할 컨벤션 부분인데요, when, then은 들여쓰기 하는 것이 어떨까요?

jamie9504 replied:
들여쓰기... 되도록 컨벤션 설정하는 법을 배우러 가겠습니다 😢

jamie9504 replied:
포맷팅 기준으로 우선 하기로 결정되었습니다!

Hamliet commented on back-end/legeno-around-here/src/test/java/wooteco/team/ittabi/legenoaroundhere/domain/PostTest.java:
```diff
@@ -0,0 +1,36 @@
+    void validateLength_OverLength_ThrownException() {
+        String overLengthInput = "aaaaaaaaaaaaaaaaaaaaa";
+        assertThatThrownBy(() -> new Post(overLengthInput))
+            .isInstanceOf(IllegalArgumentException.class);
```
hasMessage로 어떤 메시지가 나오는지 코드상으로 보여주는 것도 좋을 것 같아요!

jamie9504 replied:
예전엔 hasMessage를 자주 썼었는데, Message의 경우 변동될 수도 있기도 해서
나중엔 점점 걷어내고 Custom Exception을 잡는 방식으로 바뀌더라구요!

이 부분에 대해서 내일 얘기해보면 좋을 것 같네요 😄

Hamliet commented on back-end/legeno-around-here/src/test/java/wooteco/team/ittabi/legenoaroundhere/acceptance/PostAcceptanceTest.java:
```diff
@@ -0,0 +1,153 @@
+
+    /**
+     * Feature: 글 관리
+     * <p>
```
`<p>`태그가 있는 이유가 있을까요?

jamie9504 replied:
😯 제거해야겠네요!

jamie9504 replied:
주석 자동 포맷팅이 `<p>`가 나오네요~ 놔두도록 할게요~!

Hamliet reviewed (commented):
고생하셨습니다 ㅎㅎ 코멘트 확인해주세요!~

yelimkim98 commented on back-end/legeno-around-here/src/main/java/wooteco/team/ittabi/legenoaroundhere/service/PostService.java:
```diff
@@ -0,0 +1,59 @@
+
+    private Post findNotDeletedPost(Long id) {
+        Post post = postRepository.findById(id)
+            .orElseThrow(() -> new IllegalArgumentException("ID에 해당하는 POST가 없습니다."));
+        if (post.isSameState(State.DELETED)) {
```
if 앞에서 엔터 한 줄 어떠신가요?! 제안입니다!

aegis1920 replied:
엔터하면 더 가독성이 좋아질 것 같네요 :) 감사합니다!

jamie9504 replied:
적용했습니다.

yelimkim98 commented on back-end/legeno-around-here/src/main/java/wooteco/team/ittabi/legenoaroundhere/domain/BaseEntity.java:
```diff
@@ -0,0 +1,59 @@
+import org.springframework.data.annotation.LastModifiedDate;
+
+@MappedSuperclass
+public abstract class BaseEntity {
```
BaseEntity 라는 이름때문에
다른 모든 주요 엔티티들도 상속을 해야할 것만 같다는 느낌이 들었어요.

Post 와 Comment 같은 몇몇 entity 들에 대한 Base Entity 라는걸 명확하게 할 수 있는 네이밍이 있으면 좋을것같네요.
제안사항이 떠오르지 않아서 그냥 의견제시로만 남겨둡니다..

aegis1920 replied:
네이밍에 대해서 조금 더 생각해보겠습니다ㅎㅎ

yelimkim98 reviewed (approved):
다른 분들이 남긴 코멘트 이외에 추가적으로 남길만한 코멘트가 많지 않네요~ 제안사항 몇가지 남긴것이라서 approve 합니다 (:

jamie9504 committed:
refactor: Custom Exception 추가

- NotExistsException : 없는 리소스에 요청(+삭제된 리소스) - 404 Return
- UserInputException : 사용자가 잘못된 값을 전달 - 400 Return

jamie9504 committed:
refactor: 변수명 변경

jamie9504 committed:
refactor: controller return 부분 줄 정렬

jamie9504 committed:
refactor: LIMIT > MAX, MIN으로 재명명

jamie9504 committed:
refactor: Entity의 불필요한 Setter 메서드 제거

jamie9504 committed:
refactor: Rest Assured Version 변경

ttkmw reviewed (approved):
수고하셨습니다~

jamie9504 committed:
refactor: 컨벤션 적용 - 엔티티 필드 사이 엔터 추가

Hamliet reviewed (approved):
피드백 반영 확인했습니다ㅎㅎ 수고하셨습니다!
